File: src/main/java/net/vidageek/games/regex/task/ValidationIfMatch.java
package net.vidageek.games.regex.task;

import static net.vidageek.games.regex.task.MatcherTargets.from;

import java.util.regex.Matcher;

import net.vidageek.games.task.JudgedTask;
import net.vidageek.games.task.status.Failed;
import net.vidageek.games.task.status.Ok;

final public class ValidationIfMatch implements GroupValidation {

	private final String matchingTarget;

	public ValidationIfMatch(final String matchingTarget) {
		this.matchingTarget = matchingTarget;
	}

	public JudgedTask judge(final String challenge, final Matcher matcher) {
		if (!matcher.find()) {
			return new Failed("A regex " + from(challenge).asHtml() + " n&atilde;o d&aacute; match em "
					+ from(matchingTarget).asHtml());
		}
		return new Ok();
	}

}

File: src/main/scala/vggames/regex/task/ValidationIfMatch.scala
package vggames.regex.task;

import java.util.regex.Matcher

import vggames.regex.task.MatcherTargets.from
import vggames.shared.task.status.{Ok, Failed}
import vggames.shared.task.JudgedTask

class ValidationIfMatch(matchingTarget : String) extends GroupValidation {

  def judge(challenge : String, matcher : Matcher) : JudgedTask = {
    if (!matcher.find()) {
      return new Failed("A regex " + from(challenge).asHtml() + " n&atilde;o d&aacute; match em "
        + from(matchingTarget).asHtml())
    }
    new Ok()
  }

}
